cli-files/go-cat/cmd/root_test.go

```diff
@@ -0,0 +1,62 @@
+package cmd
+
+import (
+	"bytes"
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var TestTable = []struct {
+	Name           string
+	Args           []string
+	ExpectedOutput string
+	ExpectedError  error
+	AssertOutput   bool
```

Comment:
I guess it makes sense in this case, I didn't want to go with empty string checking just in case there's a test scenario with empty string assertion. It's a bit inconvenient that go doesn't allow nil to be assigned to string variables.

Reply:
Yeah, you definitely don't want to treat the empty string the same as not expecting a string at all - Go's belief that "default values should be useful" is, in my opinion, not a great piece of programming language design. Still, we often use pointers to indicate that something _may_ be `nil` (even though it's not unambiguous that that's what's being expressed by using a pointer).

Reply:
updated this to use pointers, thank you